bson/flexible_bson/writer.py

```diff
@@ -0,0 +1,288 @@
+# Copyright 2018-present MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""BSON (Binary JSON) encoding.
+
+"""
+
+from collections import deque
+from enum import Enum
```

Comment:
This doesn't exist in Python 2.7, and I'd rather not add a required dependency. You can use trivial classes with class level variables instead.